charts/k8ssandra/templates/stargate/service.yaml

```diff
@@ -1,8 +1,9 @@
 {{- if .Values.stargate.enabled }}
+{{- $datacenterName := (index .Values.cassandra.datacenters 0).name -}}
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ .Release.Name }}-k8ssandra-stargate-service
+  name: {{ .Release.Name }}-{{ $datacenterName }}-stargate-service
```

Comment:
When we create a separate chart for Stargate it will be easier (hopefully) to standarize on naming. This will just become:

```
name: {{ include "k8ssandra-common.fullname" . }}
```